backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/CombinedWalletClientImpl.kt

```
@@ -135,20 +167,41 @@ class CombinedWalletClientImpl private constructor(
     private suspend fun remove(serviceMode: ServiceMode) =
         withContext(Dispatchers.Default) { cache.remove(serviceMode) }
 
-    private suspend fun getOrCreate(serviceMode: ServiceMode) =
-        withContext(Dispatchers.Default) { cache.getOrPut(serviceMode) { create() } }
+    private suspend fun getOrCreate(serviceMode: ServiceMode): PartialWalletClient? =
+        withContext(Dispatchers.Default) {
+            val partialClient = cache[serviceMode]
+            if (partialClient == null) {
+                val newClient = create()
+                if (newClient != null) {
+                    cache[serviceMode] = newClient
+                    newClient
+                } else {
+                    null
+                }
+            } else {
+                partialClient
+            }
+        }
 
-    private suspend fun create() = torClient.createWalletClient("https://${endpoint.host}:${endpoint.port}")
+    @Suppress("TooGenericExceptionCaught")
+    private suspend fun create(): PartialTorWalletClient? =
+        try {
+            torClient?.createWalletClient("https://${endpoint.host}:${endpoint.port}")
+        } catch (_: Exception) {
```

Does this `catch` make sense? AFAICT all the new code is doing when this triggers, is hiding the original error, and replacing it with `UninitializedTorClientException`. But we only want `UninitializedTorClientException` to be thrown when `torClient` is null (i.e. when Tor initialization in `Synchronizer` failed).

I think undoing this `catch` and allowing errors from `torClient.createWalletClient()` to be thrown as before is what we want (and then the caller's exception handling will handle it as before, just with the additional new `UninitializedTorClientException` case).

---

also it is in this case making the functions return Response.Error.OverTor

---

I've made this function catch an exception and rethrow UninitializedTorClientException instead of returning null. The UninitializedTorClientException will then be converted to Response.Error.OverTor so that the called will need to handle only this error but not both sealed interface result and exceptions